src/Features/Core/Portable/ExternalAccess/UnitTesting/API/NewUnitTestingIncrementalAnalyzerProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.SolutionCrawler;
+using Microsoft.CodeAnalysis.Host;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api;
+
+internal sealed partial class NewUnitTestingIncrementalAnalyzerProvider : IUnitTestingIncrementalAnalyzerProvider
+{
+    private readonly string? _workspaceKind;
+    private readonly SolutionServices _services;
+    private readonly INewUnitTestingIncrementalAnalyzerProviderImplementation _incrementalAnalyzerProvider;
+
+    private IUnitTestingIncrementalAnalyzer? _lazyAnalyzer;
+
+    private NewUnitTestingIncrementalAnalyzerProvider(
+        string? workspaceKind,
+        SolutionServices services,
+        INewUnitTestingIncrementalAnalyzerProviderImplementation incrementalAnalyzerProvider)
+    {
+        _workspaceKind = workspaceKind;
+        _services = services;
+        _incrementalAnalyzerProvider = incrementalAnalyzerProvider;
+    }
+
+    // NOTE: We're currently expecting the analyzer to be singleton, so that
+    //       analyzers returned when calling this method twice would pass a reference equality check.
+    //       One instance should be created by SolutionCrawler, another one by us, when calling the
+    //       UnitTestingSolutionCrawlerServiceAccessor.Reanalyze method.
+    public IUnitTestingIncrementalAnalyzer CreateIncrementalAnalyzer()
+        => _lazyAnalyzer ??= new NewUnitTestingIncrementalAnalyzer(_incrementalAnalyzerProvider.CreateIncrementalAnalyzer());
+
+    public void Reanalyze()
+    {
+        var solutionCrawlerService = _services.GetService<IUnitTestingSolutionCrawlerService>();
+        solutionCrawlerService?.Reanalyze(
+            _workspaceKind, _services, this.CreateIncrementalAnalyzer(), projectIds: null, documentIds: null, highPriority: false);
+    }
+
+    public static NewUnitTestingIncrementalAnalyzerProvider? TryRegister(string? workspaceKind, SolutionServices services, string analyzerName, INewUnitTestingIncrementalAnalyzerProviderImplementation provider)
```

Comment:
Would this end up exposing SolutionServices to UT? Looks like the old provider type did not expose this -
http://index/?rightProject=Microsoft.CodeAnalysis.Features&file=ExternalAccess%5cUnitTesting%5cAPI%5cUnitTestingIncrementalAnalyzerProvider.cs&line=39

Reply:
yes.  this is new, and expected.  SolutionServices are a public, immutable, API that roslyn wants ourselves (and 2nd/3rd parties) to use instead of passing around Worskpaces.  Workspaces are big mutable bags that are unsafe to pass around.  We are also removing the Workspace entirely from OOP, so it has no concept hter.e  So this follows that new path we're taking :)

Reply:
Can we drop the param and use `_services` instead?

Reply:
Ah I didn't realize SolutionServices is public. Sounds good!